### ISSUE TITLE
`impl Debug` for Many Types

### ISSUE THREAD

**Issue by nixpulvis** [neon-bindings/neon#269]
It would help a lot in learning neon if I could type:

```rust
println!("{:?}", call.arguments);
// or even...
println!("{:?}", call);
```

I'm sure there are a few more types that need `Debug` as well.

**Comment by dherman** [comment]
I agree they should implement `Debug` although I'm not sure how useful we can make the output -- in particular I'm trying to figure out whether it's dangerous to actually touch the JS VM at all. If we can't, then we'd only be able to say some very generic (and not super useful) information like `struct Arguments {}`.

**Comment by kjvalencik** [comment]
In addition to `Debug`, we should eagerly implement other common traits.

https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits
